cranelift-wasm/src/sections_translator.rs

```diff
@@ -338,3 +338,23 @@ pub fn parse_data_section<'data>(
 
     Ok(())
 }
+
+/// Parses the Name section of the wasm module.
+pub fn parse_name_section<'data>(
+    names: NameSectionReader<'data>,
+    environ: &mut dyn ModuleEnvironment<'data>,
+) -> WasmResult<()> {
+    for subsection in names {
+        let function_names = match subsection? {
+            wasmparser::Name::Function(f) => f,
```

Comment:
Do you wish to handle the module name subsection too, or would you prefer that we did it in a subsequent PR?

Reply:
I went ahead and added some more changes to handle both the module name and local name subsections, since this PR isn't too large to begin with :slightly_smiling_face:

Reply:
>> Do you wish to handle the module name subsection too

> more changes to handle both the module name and local name subsections

While local name subsection is wasm specific (and I think not really useful in general case), not sure how useful module name will be.

Reply:
Makes sense! At this point that code has been removed, and function names are the only information parsed out. :slightly_smiling_face: